build.sbt

```diff
@@ -1,31 +1,52 @@
 import Dependencies._
 import org.scalafmt.sbt.ScalafmtPlugin.autoImport.scalafmtOnCompile
 
-ThisBuild / scalaVersion := "2.12.13"
+lazy val scala212 = "2.12.13"
+lazy val scala213 = "2.13.5"
+lazy val supportedScalaVersions = List(scala212, scala213)
+
+
+ThisBuild / scalaVersion := scala212
 ThisBuild / version := "0.1.0-SNAPSHOT"
 ThisBuild / organization := "org.virtuslab"
 ThisBuild / organizationName := "VirtusLab"
 Global / onChangedBuildSource := ReloadOnSourceChanges
 
+lazy val commonSettings = Seq(
+  scalafmtOnCompile := true,
+  scalacOptions ++= Seq(
+    "-deprecation",
+    "-encoding",
+    "UTF-8",
+    "-feature",
+    "-language:_",
+    "-Xfatal-warnings",
+    "-Xlog-reflective-calls",
+    "-Xlint:_",
+    "-Ybackend-parallelism",
+    "8",
+    "-Ywarn-dead-code",
+    "-Ywarn-unused:-imports,_",
+    "-unchecked"),
+  libraryDependencies ++= deps
+)
+
+
 lazy val root = (project in file("."))
+  .aggregate(core.projectRefs ++ codex.projectRefs: _*)
   .settings(
     name := "akka-safer-serializer",
-    scalafmtOnCompile := true,
-    scalacOptions ++= Seq(
-      "-deprecation",
-      "-encoding",
-      "UTF-8",
-      "-feature",
-      "-language:_",
-      "-Xfatal-warnings",
-      "-Xlog-reflective-calls",
-      "-Xlint:_",
-      "-Ybackend-parallelism",
-      "8",
-      "-Ywarn-dead-code",
-      "-Ywarn-unused:-imports,_",
-      "-unchecked"),
-    libraryDependencies ++= deps
+    crossScalaVersions := Nil,
+    publish / skip := true
   )
 
+lazy val core = (projectMatrix in file("core"))
+  .dependsOn(codex)
```

Comment:
I guess the name `core` isn't adequate (can be misleading, I'd intuitively say that it's rather non-core that should depend on core, not the other way round)... maybe let's use the near-target names `borer-akka-serializer` and `borer-extra-codecs` then for the modules

Reply:
Now that I think about it, the serializer doesn't really depend on codecs, test for it do.